File: src/types/github.ts
export interface Repository {
  id: number;
  name: string;
  full_name: string;
  description: string | null;
  html_url: string;
  homepage: string | null;
  stargazers_count: number;
  forks_count: number;
  open_issues_count: number;
  language: string | null;
  license: License | null;
  topics: string[];
  updated_at: string;
  created_at: string;
  added_to_list_at: string | null;
  category?: string;
}

export interface License {
  key: string;
  name: string;
  spdx_id: string | null;
  url: string | null;
}

export interface ScanProgress {
  total: number;
  current: number;
  currentRepo: string;
}

export interface ScanResult {
  repositories: Repository[];
  totalCount: number;
  scanDate: string;
}

export interface SavedList {
  id: string;
  name: string;
  url: string;
  repositories: Repository[];
  lastScanned: string;
  repositoryCount: number;
}

File: src-tauri/src/models/github.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Repository {
    pub id: u64,
    pub name: String,
    pub full_name: String,
    pub description: Option<String>,
    pub html_url: String,
    pub homepage: Option<String>,
    pub stargazers_count: u32,
    pub forks_count: u32,
    pub open_issues_count: u32,
    pub language: Option<String>,
    pub license: Option<License>,
    pub topics: Vec<String>,
    pub updated_at: String,
    pub created_at: String,
    pub added_to_list_at: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct License {
    pub key: String,
    pub name: String,
    pub spdx_id: Option<String>,
    pub url: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ScanProgress {
    pub total: u32,
    pub current: u32,
    pub current_repo: String,
}
